### PR TITLE
[terra-clinical-data-grid] Update Cell to have role undefined for non interactable cell

### LINKED ISSUE #703: [terra-clinical-data-grid] Add accessibility for empty cells
# Feature Request

## Description

Selectable data grid cells which do not have content currently get flagged for [accessibility violations](https://dequeuniversity.com/rules/axe/3.5/button-name?application=axeAPI).  Since empty cells are common, we would like a method by which we can be compliant with the accessibility standard. Perhaps an aria-label option in the cells properties.

## Additional Context / Screenshots

This screenshot displays a scenario where there is no text in a cell, but they violate the accessibility rules.
accessibility1

## @ Mentions

I've spoken briefly with @tbiethman about this enhancement.

### PR DESCRIPTION
**Pull request by pranav300**
### Summary

This Pr fixes the accessibility issue for selectable empty cells by making the role for cells as undefined.

Closes #703 

### Deployment Link

https://terra-clinical-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Review by ShettyAkarsh — APPROVED** [review]

File: packages/terra-clinical-data-grid/tests/jest/subcomponents/__snapshots__/HeaderCell.test.jsx.snap [hunk 1]
```diff
@@ -32,7 +32,6 @@ exports[`HeaderCell Snapshot Tests correctly applies the theme context className
       >
         <div
           className="content"
-          role="button"
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/tests/jest/subcomponents/__snapshots__/HeaderCell.test.jsx.snap:35]: The title says: "Update Cell to have role undefined for empty cell", so shouldn't this role of buttons till apply?

File: packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx [hunk 2]
```diff
@@ -133,7 +135,7 @@ class Cell extends React.Component {
         aria-selected={isSelected ? true : undefined}
       >
         <div
-          role="button"
+          role={isSelectable ? 'button' : undefined}
```
Comment (emilyrohrbough) [inline packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx:138]: if we are adding this role based on child content, should we be validating the children count?
```
React.Children.count(children) > 0

File: packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx [hunk 3]
```diff
@@ -64,6 +64,8 @@ const propTypes = {
   selectableRefCallback: PropTypes.func,
 };
 
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/no-noninteractive-tabindex */
```
Comment (mjhenkes) [inline packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx:68]: can we remove these?

File: packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx (hunk 2 above)
Comment (tbiethman) [reply packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx:138]: Let's let the implementer disable/enable the cells like they have been and just make sure our internal attributes match up. This implementation should be fine, but the CHANGELOG needs to be updated.

File: packages/terra-clinical-data-grid/tests/jest/subcomponents/__snapshots__/HeaderCell.test.jsx.snap (hunk 1 above)
Comment (pranav300) [reply packages/terra-clinical-data-grid/tests/jest/subcomponents/__snapshots__/HeaderCell.test.jsx.snap:35]: Updated the title.

File: packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx (hunk 2 above)
Comment (pranav300) [reply packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx]: Updated the CHANGELOG entry here https://github.com/cerner/terra-clinical/pull/706/commits/77a864a57d2fbb2fdc523817326c1b81f69d216a

File: packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx (hunk 3 above)
Comment (pranav300) [reply packages/terra-clinical-data-grid/src/subcomponents/Cell.jsx]: I was able to remove the `eslint-disable` for the tabIndex. But for `jsx-a11y/no-static-element-interactions` linter throws up if we don't provide any specific role.

**Review by emilyrohrbough — APPROVED** [review]

**Review by yuderekyu — APPROVED** [review]

**Review by benbcai — APPROVED** [review]

**Review by tbiethman — APPROVED** [review]

**Comment by mjhenkes** [comment]
@pranav300, Sorry i merged this too soon, we need a11y Review on it. I'll get it reverted.

**Pull request merged**